cpp/src/arrow/compute/exec/test_util.cc

```diff
@@ -459,5 +460,44 @@ void PrintTo(const Declaration& decl, std::ostream* os) {
   *os << "}";
 }
 
+std::shared_ptr<Table> MakeRandomTable(TableGenerationProperties properties) {
+  int total_columns = properties.num_columns + 2;
+  std::vector<std::shared_ptr<Array>> columns;
+  columns.reserve(total_columns);
+  arrow::FieldVector field_vector = arrow::FieldVector();
+  field_vector.reserve(total_columns);
+
+  field_vector.push_back(std::make_shared<Field>("time", int64()));
+  field_vector.push_back(std::make_shared<Field>("id", int32()));
+
+  int num_rows = 0;
```

Comment:
You could also just do `int num_rows = time_column.size()` after the for-loop.  Or you could precompute this (`((properties.end - properties.start) / properties.time_frequency) * properties.num_ids`) but it works how you have it.